### PR TITLE
Additional Diagnostics, SUNDIALS support, enable/disable precession option

### PR DESCRIPTION
**Pull request by ajnonaka**

### CODE REVIEW

**Comment by prkkumar** [comment]
Sundials works for now only on CPUs. @jclepelch You can test different time integrator options now on cpus. I will look into the CUDA build issue

File: Exec/inputs_compare_std3 [hunk 1]
```diff
@@ -33,8 +33,9 @@ exchange_parser(x,y,z) = " (x>0.)*(x<8.4e-8)*(y>0.)*(y<8.4e-8)*(z>0.)*(z<8.4e-8)
 anisotropy_parser(x,y,z) = " (x>0.)*(x<8.4e-8)*(y>0.)*(y<8.4e-8)*(z>0.)*(z<8.4e-8)* 40212.16"
 DMI_parser(x,y,z) = "0."
 
+precession = 1 
 demag_coupling = 1
-FFT_solver = 1
+FFT_solver = 0
```
Comment (ajnonaka) [inline Exec/inputs_compare_std3:38]: I've found lately solver 1 runs much faster on perlmutter (and solver 0 sometimes randomly fails, not sure why).  Can you change this back in another PR?

File: Exec/inputs_compare_std4 [hunk 2]
```diff
@@ -4,7 +4,7 @@ max_grid_size_y = 42
 max_grid_size_z = 1
 
 dt = 5.e-15  # MATLAB code uses 5.e-6 nanoseconds
-nsteps = 400000
+nsteps = 10
```
Comment (ajnonaka) [inline Exec/inputs_compare_std4:7]: can you change this back to 400,000 in another PR?

File: Exec/inputs_compare_std4 [hunk 3]
```diff
@@ -51,10 +52,10 @@ DMI_coupling = 0
 
 # INTEGRATION
 
-TimeIntegratorOption = 1 #Forward Euler
+#TimeIntegratorOption = 1 #Forward Euler
 #TimeIntegratorOption = 2 #Predictor-corrector
 #TimeIntegratorOption = 3 #2nd order artemis way
-#TimeIntegratorOption = 4 #amrex/sundials backend integrators
+TimeIntegratorOption = 4 #amrex/sundials backend integrators
```
Comment (ajnonaka) [inline Exec/inputs_compare_std4:58]: And change this back to 1 in the PR

File: Exec/inputs_performance [hunk 4]
```diff
@@ -33,6 +33,7 @@ exchange_parser(x,y,z) =   "(x>0.)*(x<512.e-9)*(y>0.)*(y<512.e-9)*(z>0.)*(z<512.
 anisotropy_parser(x,y,z) = "(x>0.)*(x<512.e-9)*(y>0.)*(y<512.e-9)*(z>0.)*(z<512.e-9) * -139.26"
 DMI_parser(x,y,z) = "0."
 
+precession = 0
```
Comment (ajnonaka) [inline Exec/inputs_performance:36]: change this back to 1 as well

**Pull request merged**